### PR TITLE
Lesson 1.6

### PR DESCRIPTION
**Pull request by secch97**
Screenshot_1

- Created a reusable input with label component: `<InputWithLabel/>`
- Refactored `<InputWithLabel/>` to use Component Composition.  Added children to said component.
- Used `useRef` hook to add autofocus to an input.
- Added and styled `remove` button. 

After assignment completion, I did the following tasks:
- Added removal animation: started approach by using `useReference `and `useEffect`. Found out that when I deleted two or more _listItems_, only the last _listItem_ removed was **truly removed** from the **todoList state** from **App.jsx**. So I had to do some modifications in order to delay the removal _once_ the removal animation was finished. 
- Added comments.
- Added new PropTypes

### CODE REVIEW

File: src/components/AddTodoForm.jsx [hunk 1]
```diff
@@ -39,24 +40,23 @@ const AddTodoForm = ({onAddTodo}) => {
...
+        inputValue={todoTitle}
+        inputName = "title"
+        inputOnChange={handleTitleChange}
+      >
+        <FontAwesomeIcon 
+          icon={["fas", "calendar-plus"]} 
+          size="xl" 
+        />
+        <span>Title: </span>
+      </InputWithLabel>
```
Comment (viniciuspietscher) [inline src/components/AddTodoForm.jsx:57]: 🥇

File: src/App.jsx [hunk 2]
```diff
@@ -25,14 +36,19 @@ const App = () => {
     setTodoList((todoList) => [...todoList, newTodo]);
   };
 
+  const handleRemoveTodo = (id) => {
+    const newTodoList = todoList.filter((toDo) => toDo.id !== id);
+    setTodoList(newTodoList);
+  }
+
```
Comment (viniciuspietscher) [inline src/App.jsx:43]: Great job! 🥇

File: src/components/InputWithLabel.jsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+  /*
+    Won't provide dependency array in order 
+    to give this side effect on mount and update
+   */
+  useEffect(() => {
+    if (inputRef.current){
+      inputRef.current.focus();
+    }
+  });
```
Comment (viniciuspietscher) [inline src/components/InputWithLabel.jsx:21]: Nice work! 🥇

**Review by viniciuspietscher — APPROVED** [review]
Great job! 🥇

File: src/components/InputWithLabel.jsx (hunk 3 above)
Comment (secch97) [reply src/components/InputWithLabel.jsx:21]: Thanks Vini!

**Pull request merged**